src/main/scala/beam/agentsim/agents/household/HouseholdActor.scala

```diff
@@ -504,4 +541,66 @@ object HouseholdActor {
     }
   }
 
+  class EmergencyHouseholdVehicleGenerator(
+    household: Household,
+    homeCoordFromPlans: Coord,
+    beamScenario: BeamScenario,
+    vehiclesAdjustment: VehiclesAdjustment,
+    defaultCategory: VehicleCategory
+  ) {
+    private val realDistribution: UniformRealDistribution = new UniformRealDistribution()
+    realDistribution.reseedRandomGenerator(beamScenario.beamConfig.matsim.modules.global.randomSeed)
+
+    private val generateEmergencyHousehold =
+      beamScenario.beamConfig.beam.agentsim.agents.vehicles.generateEmergencyHouseholdVehicleWhenPlansRequireIt
+
+    def createVehicle(personId: Id[Person], vehicleIndex: Int, category: VehicleCategory): Option[BeamVehicle] = {
+      val vehicleTypeMaybe =
+        if (generateEmergencyHousehold && defaultCategory == category) {
+          category match {
+            case VehicleCategory.Car =>
+              vehiclesAdjustment
+                .sampleVehicleTypesForHousehold(
+                  1,
+                  VehicleCategory.Car,
+                  household.getIncome.getIncome,
+                  household.getMemberIds.size(),
+                  householdPopulation = null,
+                  homeCoordFromPlans,
+                  realDistribution
+                )
+                .headOption
+                .orElse {
+                  beamScenario.vehicleTypes.get(
+                    Id.create(
+                      beamScenario.beamConfig.beam.agentsim.agents.vehicles.dummySharedCar.vehicleTypeId,
+                      classOf[BeamVehicleType]
+                    )
+                  )
+                }
+            case VehicleCategory.Bike =>
+              beamScenario.vehicleTypes
+                .get(
+                  Id.create(
+                    beamScenario.beamConfig.beam.agentsim.agents.vehicles.dummySharedBike.vehicleTypeId,
+                    classOf[BeamVehicleType]
+                  )
+                )
+            case _ => None
```

Comment:
Should we log a warning here with the type attempted?

Reply:
okay